### DISCUSSION TITLE
Multi-site handling

### DISCUSSION THREAD

**Discussion by ArtDepartmentMJ** [studioespresso/craft-scout#196] [Q&A]
Great work on v2 @riasvdv 

I've been playing around with muti-sites and even though you refer to them in your code comments, I'm wondering of it might be an idea to include an example in the readme!? 

Here's my solution that means you don't have to create a specific index each time a new site is added:

config/scout.php
````
return [
    'indices' => [] + siteIndices(),
];

/**
 * Iterate over sites and create index for each
 *
 * @return array
 * @throws Exception
 */
function siteIndices()
{
    $indices = [];
    foreach (Craft::$app->getSites()->getAllSites() as $site) {
        $indices[] = \rias\scout\ScoutIndex::create($site->handle . '_Search')
            ->elementType(\craft\elements\Entry::class)
            ->criteria(function (\craft\elements\db\EntryQuery $query) use ($site) {
                return $query->siteId($site->id)->section('pages');
            })
            ->transformer(function (\craft\elements\Entry $entry) {
                return [
                    'title' => $entry->title
                ];
            });
    }
    return $indices;
}

````

**Comment by riasvdv** [comment]
That's a nice way to handle multiple sites, I'll leave it open here for now as I don't want to overcrowd the README too much, if we have more of these I could split them to a different markdown file

**Comment by timkelty** [comment]
FWIW - you can query `*` or an array on sites now in Craft, so if you didn't need them in different algolia indices, you could do this in one criteria.

**Comment by epapada** [comment]
@ArtDepartmentMJ Hi! I am using the same solution for indexing our multi-site setup, however when I try to enter scout settings from the CP I get this error `Cannot redeclare siteIndices()` - are you having the same issue?

**Comment by epapada** [comment]
@ArtDepartmentMJ I ended up placing the function before the return array (top of the file) and checking if the function exists before declaring it - not sure if that is a proper solution.

```php
<?php
if (!function_exists('siteIndices')) {
    function siteIndices()
    {
```
